### PR TITLE
chore: Convert project permissions TestCase to normal test function

### PR DESCRIPTION
**Pull request by zachaysan**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

I'm working through re-writing the TestCase and APITestCase classes into normal test functions. I rewrote the existing classes and replaced with fixtures instead of recreating everything. There are other PRs that are similar to this one.

## How did you test this code?

I manually changed the tests one at a time to refactor them as I went then ran the full test suite afterwards.

### CODE REVIEW

**Comment by zachaysan** [comment]
@matthewelwell I tagged you for this PR since we talked about the removal of the dead code which is part of this PR.

File: api/tests/types.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 WithProjectPermissionsCallable = Callable[
-    [list[str], Project | None], UserProjectPermission
+    [list[str], int | None], UserProjectPermission
+]
+WithOrganisationPermissionsCallable = Callable[
+    [list[str], int | None], UserOrganisationPermission
 ]
 WithEnvironmentPermissionsCallable = Callable[
-    [list[str], Project | None], UserEnvironmentPermission
+    [list[str], int | None, bool], UserEnvironmentPermission
```
Comment (matthewelwell) [inline api/tests/types.py]: I think this fix was added to the wrong type here? I think it's just the `WithProjectPermissionsCallable` that currently accepts the `admin` boolean?

**Review by matthewelwell — APPROVED** [review]
Approved with one minor comment regarding the typing fix.

File: api/tests/types.py (hunk 1 above)
Comment (zachaysan) [reply api/tests/types.py]: Wow that's a great catch during a PR review! Thank you!!

Comment (zachaysan) [reply api/tests/types.py]: Also, if we were using mypy these types of things would be caught more automatically.

**Pull request merged**